node/modules/storageminer_idxprov.go

```diff
@@ -93,13 +93,25 @@ func IndexProvider(cfg config.IndexProviderConfig) func(params IdxProv, marketHo
 			// The extra data is required by the lotus-specific index-provider gossip message validators.
 			ma := address.Address(maddr)
 			opts = append(opts,
-				engine.WithPublisherKind(engine.DataTransferPublisher),
 				engine.WithDataTransfer(dtV1ToIndexerDT(dt, func() ipld.LinkSystem {
 					return *e.LinkSystem()
 				})),
 				engine.WithExtraGossipData(ma.Bytes()),
 				engine.WithTopic(t),
 			)
+
+			// Advertisements can be served over the data transfer protocol
+			// (on graphsync) or over HTTP
+			if cfg.HttpEndpoint.PublicIP != "" {
```

Comment:
minor: Worth defining a dedicated flag for HTTP publishing? I am trying to think how using the presence of HTTP endpoint as a flag can be error prone.